# Notebook for running all the examples in the examples folder

This notebook also serves as a test of the ras_commander package using the HEC-RAS Project Examples provided by HEC.

Example project names can be found in the example_projects.csv file after the examples are run.

To use a different example, change the project name in the examples and run them directly or run this notebook.  


The notebook format provides a convenient way to run all the examples to ensure that no features are broken after code revisions.  


-----

### This example notebook should take around 20-60 minutes to run.  

This table is a valuable reference for formulating examples and anticipating script behavior.

| Project           | Type     | Existing Numbers | Next Available |
|-------------------|----------|-------------------|----------------|
| Balde Eagle Creek | Plan     | 01, 02            | 03             |
|                   | Flow     | 01, 02            | 03             |
|                   | Unsteady | 02                | 01, 03         |
|                   | Geometry | 01                | 02             |
| Muncie            | Plan     | 01, 03, 04        | 02, 05         |
|                   | Flow     | 01                | 02             |
|                   | Unsteady | 01                | 02             |
|                   | Geometry | 01, 02, 04        | 03, 05         |

Notes:
1. For Balde Eagle Creek's Unsteady entries, 01 is available before 03. The library should use the lowest available number when creating new entries.
2. For Muncie's Plan entries, both 02 and 05 are available. The library should use 02 first.
3. For Muncie's Geometry entries, 03 is available before 05.

When writing examples or scripts:
1. Use existing numbers when referencing current plans, flows, unsteady setups, or geometries.
2. When creating new entries (e.g., cloning), use the "Next Available" number and then increment for subsequent creations.
3. After operations that create new entries, verify that the ras object has been updated to reflect these changes.

This approach will help ensure that examples run as expected and that we're always working with valid, up-to-date data in our scripts.

-----

# RAS-Commander (ras_commander) library examples

### The Examples Below Have Been Tested and Verified.

-----

In [1]:
#!pip install pandas pathlib requests psutil h5py matplotlib scipy

In [2]:
# Example 1: Import and execute 01_project_initialization.py
print("Executing 01_project_initialization.py:")
%run 01_project_initialization.py
print("\n")

Executing 01_project_initialization.py:


2024-09-27 12:44:19,120 - INFO - Example projects folder: c:\SCRATCH\ras-commander\examples\example_projects
2024-09-27 12:44:19,121 - WARNING - No existing example projects zip file found.
2024-09-27 12:44:19,122 - INFO - No example projects zip file found. Downloading...
2024-09-27 12:44:19,122 - INFO - Getting example projects for version 6.5
2024-09-27 12:44:19,123 - INFO - Downloading HEC-RAS Example Projects from https://github.com/HydrologicEngineeringCenter/hec-downloads/releases/download/1.0.31/Example_Projects_6_5.zip. 
The file is over 400 MB, so it may take a few minutes to download....
2024-09-27 12:44:23,139 - INFO - Downloaded to c:\SCRATCH\ras-commander\examples\Example_Projects_6_5.zip
2024-09-27 12:44:23,140 - INFO - Found zip file: c:\SCRATCH\ras-commander\examples\Example_Projects_6_5.zip
2024-09-27 12:44:23,141 - INFO - Extracting folder structure from zip file...
2024-09-27 12:44:23,150 - INFO - Extracted 66 projects.
2024-09-27 12:44:23,153 - INFO - Saved project

Example Set 1: Using the default global 'ras' object
-----------------------------------------------------
Step 1: Initializing with global RAS instance

Global RAS Instance (Bald Eagle Creek) Project Data:
Project Name: BaldEagle
Project Folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
PRJ File: C:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek\BaldEagle.prj
HEC-RAS Executable Path: C:\Program Files (x86)\HEC\HEC-RAS\6.5\Ras.exe

Plan Files DataFrame:
  plan_number                                          full_path  \
0          01  c:\SCRATCH\ras-commander\examples\example_proj...   
1          02  c:\SCRATCH\ras-commander\examples\example_proj...   

  computation_interval dss_file flow_file friction_slope_method geom_file  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  mapping_interval                     plan_title program_v

-----

In [3]:
import time

# wait 10 seconds
time.sleep(10)

In [4]:
# Example 2: Import and execute 02_plan_operations.py
print("Executing 02_plan_operations.py:")
%run 02_plan_operations.py
print("\n")

2024-09-27 12:44:34,602 - INFO - Example projects folder: c:\SCRATCH\ras-commander\examples\example_projects
2024-09-27 12:44:34,604 - INFO - Found zip file: c:\SCRATCH\ras-commander\examples\Example_Projects_6_5.zip
2024-09-27 12:44:34,606 - INFO - Loading project data from CSV...
2024-09-27 12:44:34,607 - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-09-27 12:44:34,608 - INFO - ----- RasExamples Extracting Project -----
2024-09-27 12:44:34,608 - INFO - Extracting project 'Balde Eagle Creek'
2024-09-27 12:44:34,608 - INFO - Project 'Balde Eagle Creek' already exists. Deleting existing folder...
2024-09-27 12:44:34,610 - INFO - Existing folder for project 'Balde Eagle Creek' has been deleted.
2024-09-27 12:44:34,638 - INFO - Successfully extracted project 'Balde Eagle Creek' to c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 12:44:34,638 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 

Executing 02_plan_operations.py:
Initial plan files:
  plan_number                                          full_path  \
0          01  c:\SCRATCH\ras-commander\examples\example_proj...   
1          02  c:\SCRATCH\ras-commander\examples\example_proj...   

  computation_interval dss_file flow_file friction_slope_method geom_file  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  mapping_interval                     plan_title program_version run_htab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  run_post_process run_sediment run_unet run_wqnet short_identifier  \
0                1            0        1         0     UnsteadyFlow   
1                1          NaN        1       NaN        SteadyRun   

                   simulation_date unet_d1_cores unet_use_exi

2024-09-27 12:45:35,607 - INFO - HEC-RAS execution completed for plan: 03
2024-09-27 12:45:35,608 - INFO - Total run time for plan 03: 60.90 seconds
2024-09-27 12:45:35,612 - INFO - Found 1 HDF entries.


Computing plan 03
Plan 03 computed successfully

Step 9: Retrieving HDF entries for the cloned plan
HDF entries for the cloned plan:
  plan_number                                          full_path  \
2          03  c:\SCRATCH\ras-commander\examples\example_proj...   

  computation_interval dss_file flow_file friction_slope_method geom_file  \
2                 2MIN      dss       u02                     2       g01   

  mapping_interval                     plan_title program_version run_htab  \
2            1HOUR  Unsteady with Bridges and Dam            5.00       -1   

  run_post_process run_sediment run_unet run_wqnet   short_identifier  \
2                1            0        1         0  UnsteadyFlow_copy   

                 simulation_date unet_d1_cores unet_use_existing_ib_tables  \
2  18FEB1999,0000,24FEB1999,0500             2                          -1   

                                    HDF_Results_Path  
2  c:\SCRATCH\ras-commander\examples\example_proj...  

Cur

-----

In [5]:
# wait 10 seconds
time.sleep(10)

In [6]:
# Example 3: Import and execute 03_geometry_operations.py
print("Executing 03_geometry_operations.py:")
%run 03_geometry_operations.py
print("\n")

2024-09-27 12:45:45,628 - INFO - Example projects folder: c:\SCRATCH\ras-commander\examples\example_projects
2024-09-27 12:45:45,629 - INFO - Found zip file: c:\SCRATCH\ras-commander\examples\Example_Projects_6_5.zip
2024-09-27 12:45:45,630 - INFO - Loading project data from CSV...
2024-09-27 12:45:45,631 - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-09-27 12:45:45,631 - INFO - ----- RasExamples Extracting Project -----
2024-09-27 12:45:45,632 - INFO - Extracting project 'Muncie'
2024-09-27 12:45:45,632 - INFO - Project 'Muncie' already exists. Deleting existing folder...
2024-09-27 12:45:45,640 - INFO - Existing folder for project 'Muncie' has been deleted.


Executing 03_geometry_operations.py:


2024-09-27 12:45:45,848 - INFO - Successfully extracted project 'Muncie' to c:\SCRATCH\ras-commander\examples\example_projects\Muncie
2024-09-27 12:45:45,849 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 12:45:45,849 - INFO - Initializing project in folder: c:\SCRATCH\ras-commander\examples\example_projects\Muncie
2024-09-27 12:45:45,850 - INFO - Using ras_instance with id: 140723289516752
2024-09-27 12:45:45,851 - INFO - HEC-RAS executable found at default path: C:\Program Files (x86)\HEC\HEC-RAS\6.5\Ras.exe
2024-09-27 12:45:45,851 - INFO - Initializing global 'ras' object via init_ras_project function.
2024-09-27 12:45:45,852 - INFO - Single .prj file found: c:\SCRATCH\ras-commander\examples\example_projects\Muncie\Muncie.prj
2024-09-27 12:45:45,855 - INFO - Initialization complete for project: Muncie
2024-09-27 12:45:45,856 - INFO - Plan entries: 3, Flow entries: 1, Unsteady entries: 1, Geometry entries: 3, Boundary conditions: 2
2024-09-27 12:45:45,856 - INFO - Pr

Initial plan files:
  plan_number                                          full_path  \
0          01  c:\SCRATCH\ras-commander\examples\example_proj...   
1          03  c:\SCRATCH\ras-commander\examples\example_proj...   
2          04  c:\SCRATCH\ras-commander\examples\example_proj...   

  computation_interval dss_file flow_file friction_slope_method geom_file  \
0                15SEC      dss       u01                     1       g01   
1                10SEC      dss       u01                     1       g02   
2                10SEC      dss       u01                     1       g04   

  mapping_interval                                plan_title program_version  \
0             5MIN                  Unsteady Multi  9-SA run            5.00   
1             5MIN            Unsteady Run with 2D 50ft Grid            5.10   
2             5MIN  Unsteady Run with 2D 50ft User n Value R            5.10   

   ... run_unet run_wqnet     short_identifier                simulation_date

2024-09-27 12:45:55,577 - INFO - HEC-RAS execution completed for plan: 02
2024-09-27 12:45:55,578 - INFO - Total run time for plan 02: 9.67 seconds
2024-09-27 12:45:55,583 - INFO - Results file for Plan number 02 exists at: c:\SCRATCH\ras-commander\examples\example_projects\Muncie\Muncie.p02.hdf
2024-09-27 12:45:55,585 - WARNING - Results file for Plan number 01 does not exist.


Computing plan 02
Plan 02 computed successfully

Step 7: Getting results paths
Results for plan 02 are located at: c:\SCRATCH\ras-commander\examples\example_projects\Muncie\Muncie.p02.hdf
No results found for plan 01




-----

In [7]:
# wait 10 seconds
time.sleep(10)

In [8]:
# Example 4: Import and execute 04_unsteady_flow_operations.py
print("Executing 04_unsteady_flow_operations.py:")
%run 04_unsteady_flow_operations.py
print("\n")

2024-09-27 12:46:05,612 - INFO - Example projects folder: c:\SCRATCH\ras-commander\examples\example_projects
2024-09-27 12:46:05,613 - INFO - Found zip file: c:\SCRATCH\ras-commander\examples\Example_Projects_6_5.zip
2024-09-27 12:46:05,613 - INFO - Loading project data from CSV...
2024-09-27 12:46:05,615 - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-09-27 12:46:05,616 - INFO - ----- RasExamples Extracting Project -----
2024-09-27 12:46:05,616 - INFO - Extracting project 'Balde Eagle Creek'
2024-09-27 12:46:05,616 - INFO - Project 'Balde Eagle Creek' already exists. Deleting existing folder...
2024-09-27 12:46:05,622 - INFO - Existing folder for project 'Balde Eagle Creek' has been deleted.
2024-09-27 12:46:05,648 - INFO - Successfully extracted project 'Balde Eagle Creek' to c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 12:46:05,649 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 

Executing 04_unsteady_flow_operations.py:
Initial plan files:
  plan_number                                          full_path  \
0          01  c:\SCRATCH\ras-commander\examples\example_proj...   
1          02  c:\SCRATCH\ras-commander\examples\example_proj...   

  computation_interval dss_file flow_file friction_slope_method geom_file  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  mapping_interval                     plan_title program_version run_htab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  run_post_process run_sediment run_unet run_wqnet short_identifier  \
0                1            0        1         0     UnsteadyFlow   
1                1          NaN        1       NaN        SteadyRun   

                   simulation_date unet_d1_cores une

2024-09-27 12:47:05,807 - INFO - HEC-RAS execution completed for plan: 03
2024-09-27 12:47:05,808 - INFO - Total run time for plan 03: 60.12 seconds


Plan 03 computed successfully




-----

In [9]:
# wait 10 seconds
time.sleep(10)

In [10]:
# Example 5: Import and execute 05_utility_functions.py
print("Executing 05_utility_functions.py:")
%run 05_utility_functions.py
print("\n")

2024-09-27 12:47:15,850 - INFO - Example projects folder: c:\SCRATCH\ras-commander\examples\example_projects
2024-09-27 12:47:15,853 - INFO - Found zip file: c:\SCRATCH\ras-commander\examples\Example_Projects_6_5.zip
2024-09-27 12:47:15,854 - INFO - Loading project data from CSV...
2024-09-27 12:47:15,856 - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-09-27 12:47:15,856 - INFO - ----- RasExamples Extracting Project -----
2024-09-27 12:47:15,856 - INFO - Extracting project 'Balde Eagle Creek'
2024-09-27 12:47:15,857 - INFO - Project 'Balde Eagle Creek' already exists. Deleting existing folder...
2024-09-27 12:47:15,862 - INFO - Existing folder for project 'Balde Eagle Creek' has been deleted.
2024-09-27 12:47:15,888 - INFO - Successfully extracted project 'Balde Eagle Creek' to c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 12:47:15,889 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 

Executing 05_utility_functions.py:
Example 1: Getting plan path
Path for plan 01 is: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek\BaldEagle.p01

Example 2: Getting geometry path
Path for geometry 01 is: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek\BaldEagle.g01

Example 3: Getting unsteady flow path
Path for unsteady flow 01 is: None

Example 4: Getting project name
Project name: BaldEagle




-----

In [11]:
# wait 10 seconds
time.sleep(10)

In [12]:
# Example 6: Import and execute 06_single_plan_execution.py
print("Executing 06_single_plan_execution.py:")
%run 06_single_plan_execution.py
print("\n")

2024-09-27 12:47:25,969 - INFO - Example projects folder: c:\SCRATCH\ras-commander\examples\example_projects
2024-09-27 12:47:25,970 - INFO - Found zip file: c:\SCRATCH\ras-commander\examples\Example_Projects_6_5.zip
2024-09-27 12:47:25,970 - INFO - Loading project data from CSV...
2024-09-27 12:47:25,971 - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-09-27 12:47:25,972 - INFO - ----- RasExamples Extracting Project -----
2024-09-27 12:47:25,972 - INFO - Extracting project 'Balde Eagle Creek'
2024-09-27 12:47:25,998 - INFO - Successfully extracted project 'Balde Eagle Creek' to c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 12:47:25,998 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 12:47:25,999 - INFO - Initializing project in folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 12:47:25,999 - INFO - Using ras_instance with id: 140723289516752
2024

Executing 06_single_plan_execution.py:
Available plans:
  plan_number                                          full_path  \
0          01  c:\SCRATCH\ras-commander\examples\example_proj...   
1          02  c:\SCRATCH\ras-commander\examples\example_proj...   

  computation_interval dss_file flow_file friction_slope_method geom_file  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  mapping_interval                     plan_title program_version run_htab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  run_post_process run_sediment run_unet run_wqnet short_identifier  \
0                1            0        1         0     UnsteadyFlow   
1                1          NaN        1       NaN        SteadyRun   

                   simulation_date unet_d1_cores unet_use_

2024-09-27 12:48:26,596 - INFO - HEC-RAS execution completed for plan: 01
2024-09-27 12:48:26,596 - INFO - Total run time for plan 01: 60.59 seconds
2024-09-27 12:48:26,599 - INFO - Using ras_object with project folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 12:48:26,626 - INFO - Copied project folder to destination: c:\SCRATCH\ras-commander\examples\example_projects\compute_test_2
2024-09-27 12:48:26,627 - INFO - Single .prj file found: c:\SCRATCH\ras-commander\examples\example_projects\compute_test_2\BaldEagle.prj
2024-09-27 12:48:26,631 - INFO - Initialization complete for project: BaldEagle
2024-09-27 12:48:26,631 - INFO - Plan entries: 2, Flow entries: 2, Unsteady entries: 1, Geometry entries: 1, Boundary conditions: 3
2024-09-27 12:48:26,633 - INFO - Plan file for Plan number 02 found at: c:\SCRATCH\ras-commander\examples\example_projects\compute_test_2\BaldEagle.p02
2024-09-27 12:48:26,633 - INFO - Running HEC-RAS from the Command Line:
2

Plan 01 executed successfully

Example 2: Executing a plan in a separate destination folder


2024-09-27 12:48:29,635 - INFO - HEC-RAS execution completed for plan: 02
2024-09-27 12:48:29,636 - INFO - Total run time for plan 02: 3.00 seconds
2024-09-27 12:48:29,639 - WARNING - Results file for Plan number 02 does not exist.
2024-09-27 12:48:29,639 - INFO - Using ras_object with project folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 12:48:29,667 - INFO - Copied project folder to destination: c:\SCRATCH\ras-commander\examples\example_projects\compute_test_3
2024-09-27 12:48:29,668 - INFO - Single .prj file found: c:\SCRATCH\ras-commander\examples\example_projects\compute_test_3\BaldEagle.prj
2024-09-27 12:48:29,673 - INFO - Initialization complete for project: BaldEagle
2024-09-27 12:48:29,674 - INFO - Plan entries: 2, Flow entries: 2, Unsteady entries: 1, Geometry entries: 1, Boundary conditions: 3
2024-09-27 12:48:29,676 - WARNING - Plan number 03 not found in the updated plan entries.
2024-09-27 12:48:29,677 - ERROR - Could not find pro

Plan 02 executed successfully in c:\SCRATCH\ras-commander\examples\example_projects\compute_test_2

Example 3: Getting results path
No results found for plan 02

Example 4: Executing a plan with cleared geometry preprocessor files
Plan 03 execution failed

Example 5: Executing a plan with a specified number of cores, overwriting compute_test_3


2024-09-27 12:48:40,158 - INFO - HEC-RAS execution completed for plan: 01
2024-09-27 12:48:40,158 - INFO - Total run time for plan 01: 10.44 seconds
2024-09-27 12:48:40,161 - INFO - Using ras_object with project folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 12:48:40,188 - INFO - Copied project folder to destination: c:\SCRATCH\ras-commander\examples\example_projects\compute_test_all_options
2024-09-27 12:48:40,189 - INFO - Single .prj file found: c:\SCRATCH\ras-commander\examples\example_projects\compute_test_all_options\BaldEagle.prj
2024-09-27 12:48:40,193 - INFO - Initialization complete for project: BaldEagle
2024-09-27 12:48:40,193 - INFO - Plan entries: 2, Flow entries: 2, Unsteady entries: 1, Geometry entries: 1, Boundary conditions: 3
2024-09-27 12:48:40,195 - INFO - Plan file for Plan number 02 found at: c:\SCRATCH\ras-commander\examples\example_projects\compute_test_all_options\BaldEagle.p02
2024-09-27 12:48:40,195 - INFO - Clearing g

Plan 01 executed successfully using 2 cores

Example 6: Executing a plan with all new options combined


2024-09-27 12:48:45,335 - INFO - HEC-RAS execution completed for plan: 02
2024-09-27 12:48:45,335 - INFO - Total run time for plan 02: 5.13 seconds


Plan 02 executed successfully with all options:
- Destination folder: c:\SCRATCH\ras-commander\examples\example_projects\compute_test_all_options
- Cleared geometry preprocessor files
- Used 4 cores




-----

In [13]:
# wait 10 seconds
time.sleep(10)

In [14]:
# Example 7: Import and execute 07_sequential_plan_execution.py
print("Executing 07_sequential_plan_execution.py:")
%run 07_sequential_plan_execution.py
print("\n")

2024-09-27 12:48:55,376 - INFO - Example projects folder: c:\SCRATCH\ras-commander\examples\example_projects
2024-09-27 12:48:55,376 - INFO - Found zip file: c:\SCRATCH\ras-commander\examples\Example_Projects_6_5.zip
2024-09-27 12:48:55,377 - INFO - Loading project data from CSV...
2024-09-27 12:48:55,378 - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-09-27 12:48:55,379 - INFO - ----- RasExamples Extracting Project -----
2024-09-27 12:48:55,379 - INFO - Extracting project 'Balde Eagle Creek'
2024-09-27 12:48:55,405 - INFO - Successfully extracted project 'Balde Eagle Creek' to c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 12:48:55,406 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 12:48:55,406 - INFO - Initializing project in folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 12:48:55,407 - INFO - Using ras_instance with id: 140723289516752
2024

Executing 07_sequential_plan_execution.py:
Removing existing folder: C:\SCRATCH\ras-commander\examples\example_projects
Available plans:
  plan_number                                          full_path  \
0          01  c:\SCRATCH\ras-commander\examples\example_proj...   
1          02  c:\SCRATCH\ras-commander\examples\example_proj...   

  computation_interval dss_file flow_file friction_slope_method geom_file  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  mapping_interval                     plan_title program_version run_htab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  run_post_process run_sediment run_unet run_wqnet short_identifier  \
0                1            0        1         0     UnsteadyFlow   
1                1          NaN        1       Na

2024-09-27 12:49:56,368 - INFO - HEC-RAS execution completed for plan: 01
2024-09-27 12:49:56,369 - INFO - Total run time for plan 01: 60.93 seconds
2024-09-27 12:49:56,371 - INFO - Successfully computed plan 01
2024-09-27 12:49:56,372 - INFO - Total run time for plan 01: 60.94 seconds
2024-09-27 12:49:56,372 - INFO - Using ras_object with project folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [AllSequential]
2024-09-27 12:49:56,374 - INFO - Plan file for Plan number 02 found at: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [AllSequential]\BaldEagle.p02
2024-09-27 12:49:56,374 - INFO - Running HEC-RAS from the Command Line:
2024-09-27 12:49:56,375 - INFO - Running command: "C:\Program Files (x86)\HEC\HEC-RAS\6.5\Ras.exe" -c "C:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [AllSequential]\BaldEagle.prj" "c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [AllSequential]\BaldEagle.p02"
2024-09-27

Sequential execution of all plans completed with overwrite_dest

Example 2: Sequential execution of specific plans with clearing geompre files and overwrite_dest


2024-09-27 12:50:58,738 - INFO - HEC-RAS execution completed for plan: 01
2024-09-27 12:50:58,739 - INFO - Total run time for plan 01: 59.53 seconds
2024-09-27 12:50:58,741 - INFO - Successfully computed plan 01
2024-09-27 12:50:58,742 - INFO - Total run time for plan 01: 59.54 seconds
2024-09-27 12:50:58,742 - INFO - Using ras_object with project folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [SpecificSequentialClearGeompre]
2024-09-27 12:50:58,743 - INFO - Plan file for Plan number 02 found at: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [SpecificSequentialClearGeompre]\BaldEagle.p02
2024-09-27 12:50:58,744 - INFO - Clearing geometry preprocessor file for single plan: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [SpecificSequentialClearGeompre]\BaldEagle.p02
2024-09-27 12:50:58,744 - WARNING - No geometry preprocessor file found for: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [Spe

Sequential execution of specific plans completed with clearing geompre files and overwrite_dest

Example 3: Clearing geompre files for specific plans
Geometry preprocessor files cleared for specific plans

Example 4: Clearing all geompre files
All geometry preprocessor files cleared




-----

In [15]:
# wait 10 seconds
time.sleep(10)

In [16]:
# Example 8: Import and execute 08_parallel_execution.py
print("Executing 08_parallel_execution.py:")
%run 08_parallel_execution.py
print("\n")

2024-09-27 12:51:11,700 - INFO - Example projects folder: c:\SCRATCH\ras-commander\examples\example_projects
2024-09-27 12:51:11,700 - INFO - Found zip file: c:\SCRATCH\ras-commander\examples\Example_Projects_6_5.zip
2024-09-27 12:51:11,701 - INFO - Loading project data from CSV...
2024-09-27 12:51:11,702 - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-09-27 12:51:11,702 - INFO - ----- RasExamples Extracting Project -----
2024-09-27 12:51:11,703 - INFO - Extracting project 'Balde Eagle Creek'
2024-09-27 12:51:11,703 - INFO - Project 'Balde Eagle Creek' already exists. Deleting existing folder...
2024-09-27 12:51:11,705 - INFO - Existing folder for project 'Balde Eagle Creek' has been deleted.
2024-09-27 12:51:11,733 - INFO - Successfully extracted project 'Balde Eagle Creek' to c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 12:51:11,733 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 

Executing 08_parallel_execution.py:
Available plans:
  plan_number                                          full_path  \
0          01  c:\SCRATCH\ras-commander\examples\example_proj...   
1          02  c:\SCRATCH\ras-commander\examples\example_proj...   

  computation_interval dss_file flow_file friction_slope_method geom_file  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  mapping_interval                     plan_title program_version run_htab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  run_post_process run_sediment run_unet run_wqnet short_identifier  \
0                1            0        1         0     UnsteadyFlow   
1                1          NaN        1       NaN        SteadyRun   

                   simulation_date unet_d1_cores unet_use_exi

2024-09-27 12:51:17,043 - INFO - HEC-RAS execution completed for plan: 02
2024-09-27 12:51:17,043 - INFO - Total run time for plan 02: 5.24 seconds
2024-09-27 12:51:17,045 - INFO - Plan 01 executed in worker 1
2024-09-27 12:52:11,146 - INFO - HEC-RAS execution completed for plan: 01
2024-09-27 12:52:11,146 - INFO - Total run time for plan 01: 59.34 seconds
2024-09-27 12:52:11,149 - INFO - Plan 02 executed in worker 2
2024-09-27 12:52:11,149 - INFO - Final destination for computed results: c:\SCRATCH\ras-commander\examples\example_projects\compute_test_parallel
2024-09-27 12:52:11,161 - ERROR - Error moving results from c:\SCRATCH\ras-commander\examples\example_projects\compute_test_parallel [Worker 1] to c:\SCRATCH\ras-commander\examples\example_projects\compute_test_parallel: [WinError 32] The process cannot access the file because it is being used by another process: 'c:\\SCRATCH\\ras-commander\\examples\\example_projects\\compute_test_parallel [Worker 1]'
2024-09-27 12:52:11,170 - I

Parallel execution of all plans results:
Plan 01: Successful
Plan 02: Successful

Plan DataFrame after parallel execution of all plans:
  plan_number                                          full_path  \
0          01  c:\SCRATCH\ras-commander\examples\example_proj...   
1          02  c:\SCRATCH\ras-commander\examples\example_proj...   

  computation_interval dss_file flow_file friction_slope_method geom_file  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  mapping_interval                     plan_title program_version run_htab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  run_post_process run_sediment run_unet run_wqnet short_identifier  \
0                1            0        1         0     UnsteadyFlow   
1                1          NaN        1       NaN

2024-09-27 12:52:14,267 - INFO - HEC-RAS execution completed for plan: 02
2024-09-27 12:52:14,268 - INFO - Total run time for plan 02: 2.94 seconds
2024-09-27 12:52:14,270 - INFO - Plan 01 executed in worker 1
2024-09-27 12:53:10,458 - INFO - HEC-RAS execution completed for plan: 01
2024-09-27 12:53:10,459 - INFO - Total run time for plan 01: 59.13 seconds
2024-09-27 12:53:10,461 - INFO - Plan 02 executed in worker 2
2024-09-27 12:53:10,461 - INFO - Final destination for computed results: c:\SCRATCH\ras-commander\examples\example_projects\compute_test_parallel_specific
2024-09-27 12:53:10,477 - ERROR - Error moving results from c:\SCRATCH\ras-commander\examples\example_projects\compute_test_parallel_specific [Worker 1] to c:\SCRATCH\ras-commander\examples\example_projects\compute_test_parallel_specific: [WinError 32] The process cannot access the file because it is being used by another process: 'c:\\SCRATCH\\ras-commander\\examples\\example_projects\\compute_test_parallel_specific [Wo

Parallel execution of specific plans results:
Plan 01: Successful
Plan 02: Successful

Example 3: Parallel execution with dynamic max_workers


2024-09-27 12:53:15,989 - INFO - HEC-RAS execution completed for plan: 02
2024-09-27 12:53:15,989 - INFO - Total run time for plan 02: 5.37 seconds
2024-09-27 12:53:15,992 - INFO - Plan 01 executed in worker 1
2024-09-27 12:54:10,088 - INFO - HEC-RAS execution completed for plan: 01
2024-09-27 12:54:10,088 - INFO - Total run time for plan 01: 59.47 seconds
2024-09-27 12:54:10,090 - INFO - Plan 02 executed in worker 2
2024-09-27 12:54:10,091 - INFO - Final destination for computed results: c:\SCRATCH\ras-commander\examples\example_projects\compute_test_parallel_dynamic
2024-09-27 12:54:10,106 - ERROR - Error moving results from c:\SCRATCH\ras-commander\examples\example_projects\compute_test_parallel_dynamic [Worker 1] to c:\SCRATCH\ras-commander\examples\example_projects\compute_test_parallel_dynamic: [WinError 32] The process cannot access the file because it is being used by another process: 'c:\\SCRATCH\\ras-commander\\examples\\example_projects\\compute_test_parallel_dynamic [Worker

Parallel execution with 4 workers and 2 cores per worker:
Plan 01: Successful
Plan 02: Successful

Results paths for dynamic execution:
  plan_number                                          full_path  \
0          01  c:\SCRATCH\ras-commander\examples\example_proj...   
1          02  c:\SCRATCH\ras-commander\examples\example_proj...   

  computation_interval dss_file flow_file friction_slope_method geom_file  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  mapping_interval                     plan_title program_version run_htab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  run_post_process run_sediment run_unet run_wqnet short_identifier  \
0                1            0        1         0     UnsteadyFlow   
1                1          NaN        1       NaN

-----

In [17]:
# wait 10 seconds
time.sleep(10)

In [18]:
# Example 9: Import and execute 09_specifying_plans.py
print("Executing 09_specifying_plans.py:")
%run 09_specifying_plans.py
print("\n")

2024-09-27 12:54:20,197 - INFO - Example projects folder: c:\SCRATCH\ras-commander\examples\example_projects
2024-09-27 12:54:20,199 - INFO - Found zip file: c:\SCRATCH\ras-commander\examples\Example_Projects_6_5.zip
2024-09-27 12:54:20,199 - INFO - Loading project data from CSV...
2024-09-27 12:54:20,200 - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-09-27 12:54:20,201 - INFO - ----- RasExamples Extracting Project -----
2024-09-27 12:54:20,201 - INFO - Extracting project 'Balde Eagle Creek'
2024-09-27 12:54:20,227 - INFO - Successfully extracted project 'Balde Eagle Creek' to c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 12:54:20,227 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 12:54:20,228 - INFO - Initializing project in folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 12:54:20,228 - INFO - Using ras_instance with id: 140723289516752
2024

Executing 09_specifying_plans.py:
Removing existing folder: C:\SCRATCH\ras-commander\examples\example_projects
Available plans:
  plan_number                                          full_path  \
0          01  c:\SCRATCH\ras-commander\examples\example_proj...   
1          02  c:\SCRATCH\ras-commander\examples\example_proj...   

  computation_interval dss_file flow_file friction_slope_method geom_file  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  mapping_interval                     plan_title program_version run_htab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  run_post_process run_sediment run_unet run_wqnet short_identifier  \
0                1            0        1         0     UnsteadyFlow   
1                1          NaN        1       NaN        

2024-09-27 12:55:19,323 - INFO - HEC-RAS execution completed for plan: 01
2024-09-27 12:55:19,324 - INFO - Total run time for plan 01: 59.06 seconds
2024-09-27 12:55:19,326 - INFO - Successfully computed plan 01
2024-09-27 12:55:19,326 - INFO - Total run time for plan 01: 59.07 seconds
2024-09-27 12:55:19,326 - INFO - All selected plans have been executed.
2024-09-27 12:55:19,327 - INFO - compute_test_mode completed.
2024-09-27 12:55:19,329 - INFO - Filtered plans to execute: ['01', '02']
2024-09-27 12:55:19,330 - INFO - Adjusted max_workers to 2 based on the number of plans: 2
2024-09-27 12:55:19,342 - INFO - Created worker folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [Worker 1]
2024-09-27 12:55:19,343 - INFO - Initializing project in folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [Worker 1]
2024-09-27 12:55:19,343 - INFO - Using ras_instance with id: 1957881636048
2024-09-27 12:55:19,344 - INFO - HEC-RAS executable found a

Sequential execution of specific plans completed

Example 2: Parallel execution of specific plans


2024-09-27 12:55:25,852 - INFO - HEC-RAS execution completed for plan: 02
2024-09-27 12:55:25,852 - INFO - Total run time for plan 02: 6.45 seconds
2024-09-27 12:55:25,854 - INFO - Plan 01 executed in worker 1
2024-09-27 12:56:18,871 - INFO - HEC-RAS execution completed for plan: 01
2024-09-27 12:56:18,871 - INFO - Total run time for plan 01: 59.47 seconds
2024-09-27 12:56:18,874 - INFO - Plan 02 executed in worker 2
2024-09-27 12:56:18,874 - INFO - Final destination for computed results: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [Computed]
2024-09-27 12:56:18,884 - ERROR - Error moving results from c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [Worker 1] to c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [Computed]: [WinError 32] The process cannot access the file because it is being used by another process: 'c:\\SCRATCH\\ras-commander\\examples\\example_projects\\Balde Eagle Creek [Worker 1]'
2024-09-27 12:56:18,8

Parallel execution of specific plans results:
Plan 01: Successful
Plan 02: Successful

Example 3: Execute all plans


2024-09-27 12:57:18,518 - INFO - HEC-RAS execution completed for plan: 01
2024-09-27 12:57:18,518 - INFO - Total run time for plan 01: 59.59 seconds
2024-09-27 12:57:18,520 - INFO - Successfully computed plan 01
2024-09-27 12:57:18,521 - INFO - Total run time for plan 01: 59.60 seconds
2024-09-27 12:57:18,521 - INFO - Using ras_object with project folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [AllPlans]
2024-09-27 12:57:18,523 - INFO - Plan file for Plan number 02 found at: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [AllPlans]\BaldEagle.p02
2024-09-27 12:57:18,523 - INFO - Running HEC-RAS from the Command Line:
2024-09-27 12:57:18,523 - INFO - Running command: "C:\Program Files (x86)\HEC\HEC-RAS\6.5\Ras.exe" -c "C:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [AllPlans]\BaldEagle.prj" "c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [AllPlans]\BaldEagle.p02"
2024-09-27 12:57:21,463 - INFO

Execution of all plans completed





-----

In [19]:
# wait 10 seconds
time.sleep(10)

In [20]:
print("Executing 10_arguments_for_compute.py:")
%run 10_arguments_for_compute.py
print("\n")

2024-09-27 12:57:31,481 - INFO - Example projects folder: c:\SCRATCH\ras-commander\examples\example_projects
2024-09-27 12:57:31,481 - INFO - Found zip file: c:\SCRATCH\ras-commander\examples\Example_Projects_6_5.zip
2024-09-27 12:57:31,482 - INFO - Loading project data from CSV...
2024-09-27 12:57:31,483 - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-09-27 12:57:31,483 - INFO - ----- RasExamples Extracting Project -----
2024-09-27 12:57:31,483 - INFO - Extracting project 'Balde Eagle Creek'
2024-09-27 12:57:31,484 - INFO - Project 'Balde Eagle Creek' already exists. Deleting existing folder...
2024-09-27 12:57:31,486 - INFO - Existing folder for project 'Balde Eagle Creek' has been deleted.
2024-09-27 12:57:31,512 - INFO - Successfully extracted project 'Balde Eagle Creek' to c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 12:57:31,513 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 

Executing 10_arguments_for_compute.py:
Available plans:
  plan_number                                          full_path  \
0          01  c:\SCRATCH\ras-commander\examples\example_proj...   
1          02  c:\SCRATCH\ras-commander\examples\example_proj...   

  computation_interval dss_file flow_file friction_slope_method geom_file  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  mapping_interval                     plan_title program_version run_htab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  run_post_process run_sediment run_unet run_wqnet short_identifier  \
0                1            0        1         0     UnsteadyFlow   
1                1          NaN        1       NaN        SteadyRun   

                   simulation_date unet_d1_cores unet_use_

2024-09-27 12:58:30,898 - INFO - HEC-RAS execution completed for plan: 01
2024-09-27 12:58:30,898 - INFO - Total run time for plan 01: 59.34 seconds
2024-09-27 12:58:30,900 - INFO - Successfully computed plan 01
2024-09-27 12:58:30,901 - INFO - Total run time for plan 01: 59.35 seconds
2024-09-27 12:58:30,901 - INFO - All selected plans have been executed.
2024-09-27 12:58:30,901 - INFO - compute_test_mode completed.
2024-09-27 12:58:30,904 - INFO - Starting the compute_test_mode...
2024-09-27 12:58:30,904 - INFO - Creating the test folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek _02_[SequentialWithArgs]...
2024-09-27 12:58:30,917 - INFO - Copied project folder to compute folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek _02_[SequentialWithArgs]
2024-09-27 12:58:30,918 - INFO - Single .prj file found: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek _02_[SequentialWithArgs]\BaldEagle.prj
2024-09-27 12:58:30,922

Plan 01 execution: Failed


2024-09-27 12:58:35,626 - INFO - HEC-RAS execution completed for plan: 02
2024-09-27 12:58:35,626 - INFO - Total run time for plan 02: 4.69 seconds
2024-09-27 12:58:35,628 - INFO - Successfully computed plan 02
2024-09-27 12:58:35,628 - INFO - Total run time for plan 02: 4.70 seconds
2024-09-27 12:58:35,629 - INFO - All selected plans have been executed.
2024-09-27 12:58:35,629 - INFO - compute_test_mode completed.
2024-09-27 12:58:35,631 - INFO - Starting the compute_test_mode...
2024-09-27 12:58:35,632 - INFO - Creating the test folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek _02_[SequentialWithArgs]...
2024-09-27 12:58:35,632 - ERROR - Compute folder 'c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek _02_[SequentialWithArgs]' exists and is not empty. Use overwrite_dest=True to overwrite.
2024-09-27 12:58:35,644 - INFO - Copied project folder to destination: c:\SCRATCH\ras-commander\examples\example_projects\parallel_results
2024-09-27 

Plan 02 execution: Failed
Sequential execution completed

If the example operates successfully (it is meant to generate an error above), you will not see this message.
Example 2: Parallel execution with various arguments


2024-09-27 12:58:40,897 - INFO - HEC-RAS execution completed for plan: 02
2024-09-27 12:58:40,897 - INFO - Total run time for plan 02: 5.18 seconds
2024-09-27 12:58:40,899 - INFO - Plan 01 executed in worker 1
2024-09-27 12:59:35,047 - INFO - HEC-RAS execution completed for plan: 01
2024-09-27 12:59:35,047 - INFO - Total run time for plan 01: 59.33 seconds
2024-09-27 12:59:35,049 - INFO - Plan 02 executed in worker 2
2024-09-27 12:59:35,050 - INFO - Final destination for computed results: c:\SCRATCH\ras-commander\examples\example_projects\parallel_results
2024-09-27 12:59:35,061 - ERROR - Error moving results from c:\SCRATCH\ras-commander\examples\example_projects\parallel_results [Worker 1] to c:\SCRATCH\ras-commander\examples\example_projects\parallel_results: [WinError 32] The process cannot access the file because it is being used by another process: 'c:\\SCRATCH\\ras-commander\\examples\\example_projects\\parallel_results [Worker 1]'
2024-09-27 12:59:35,071 - INFO - Removed worker

Parallel execution results:
Plan 01: Successful
Plan 02: Successful

Example 3: Single plan execution with specific arguments


2024-09-27 12:59:40,278 - INFO - HEC-RAS execution completed for plan: 02
2024-09-27 12:59:40,279 - INFO - Total run time for plan 02: 5.17 seconds


Single plan execution: Successful




In [21]:
# wait 10 seconds
time.sleep(10)

In [22]:
print("Executing 12_plan_set_execution.py:")
%run 12_plan_set_execution.py
print("\n")

2024-09-27 12:59:50,300 - INFO - Example projects folder: c:\SCRATCH\ras-commander\examples\example_projects
2024-09-27 12:59:50,300 - INFO - Found zip file: c:\SCRATCH\ras-commander\examples\Example_Projects_6_5.zip
2024-09-27 12:59:50,301 - INFO - Loading project data from CSV...
2024-09-27 12:59:50,302 - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-09-27 12:59:50,302 - INFO - ----- RasExamples Extracting Project -----
2024-09-27 12:59:50,303 - INFO - Extracting project 'Balde Eagle Creek'
2024-09-27 12:59:50,303 - INFO - Project 'Balde Eagle Creek' already exists. Deleting existing folder...
2024-09-27 12:59:50,305 - INFO - Existing folder for project 'Balde Eagle Creek' has been deleted.
2024-09-27 12:59:50,333 - INFO - Successfully extracted project 'Balde Eagle Creek' to c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 12:59:50,334 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 

Executing 12_plan_set_execution.py:
Available plans:
  plan_number                                          full_path  \
0          01  c:\SCRATCH\ras-commander\examples\example_proj...   
1          02  c:\SCRATCH\ras-commander\examples\example_proj...   

  computation_interval dss_file flow_file friction_slope_method geom_file  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  mapping_interval                     plan_title program_version run_htab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  run_post_process run_sediment run_unet run_wqnet short_identifier  \
0                1            0        1         0     UnsteadyFlow   
1                1          NaN        1       NaN        SteadyRun   

                   simulation_date unet_d1_cores unet_use_exi

2024-09-27 12:59:50,496 - INFO - Created worker folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [Worker 1]
2024-09-27 12:59:50,497 - INFO - Initializing project in folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [Worker 1]
2024-09-27 12:59:50,497 - INFO - Using ras_instance with id: 1958156474128
2024-09-27 12:59:50,497 - INFO - HEC-RAS executable found at specified path: C:\Program Files (x86)\HEC\HEC-RAS\6.5\Ras.exe
2024-09-27 12:59:50,498 - INFO - Single .prj file found: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [Worker 1]\BaldEagle.prj
2024-09-27 12:59:50,503 - INFO - Initialization complete for project: BaldEagle
2024-09-27 12:59:50,504 - INFO - Plan entries: 7, Flow entries: 2, Unsteady entries: 1, Geometry entries: 6, Boundary conditions: 3
2024-09-27 12:59:50,504 - INFO - Project initialized. ras_instance project folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [Work


Plan set execution results:
  plan_number geom_number  execution_success
0          03          02               True
1          04          03               True
2          05          04               True
3          06          05               True
4          07          06               True




In [23]:
# wait 10 seconds
time.sleep(10)

In [24]:
print("Executing 13_multiple_project_operations.py:")
%run 13_multiple_project_operations.py
print("\n")

2024-09-27 13:01:15,919 - INFO - Example projects folder: c:\SCRATCH\ras-commander\examples\example_projects
2024-09-27 13:01:15,920 - INFO - Found zip file: c:\SCRATCH\ras-commander\examples\Example_Projects_6_5.zip
2024-09-27 13:01:15,920 - INFO - Loading project data from CSV...
2024-09-27 13:01:15,922 - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-09-27 13:01:15,922 - INFO - ----- RasExamples Extracting Project -----
2024-09-27 13:01:15,923 - INFO - Extracting project 'Balde Eagle Creek'
2024-09-27 13:01:15,923 - INFO - Project 'Balde Eagle Creek' already exists. Deleting existing folder...
2024-09-27 13:01:15,926 - INFO - Existing folder for project 'Balde Eagle Creek' has been deleted.
2024-09-27 13:01:15,953 - INFO - Successfully extracted project 'Balde Eagle Creek' to c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 13:01:15,953 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 

Executing 13_multiple_project_operations.py:


2024-09-27 13:01:16,156 - INFO - Successfully extracted project 'Muncie' to c:\SCRATCH\ras-commander\examples\example_projects\Muncie
2024-09-27 13:01:16,157 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 13:01:16,157 - INFO - Initializing project in folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 13:01:16,157 - INFO - Using ras_instance with id: 140723289516752
2024-09-27 13:01:16,158 - INFO - HEC-RAS executable found at default path: C:\Program Files (x86)\HEC\HEC-RAS\6.5\Ras.exe
2024-09-27 13:01:16,158 - INFO - Initializing global 'ras' object via init_ras_project function.
2024-09-27 13:01:16,159 - INFO - Single .prj file found: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek\BaldEagle.prj
2024-09-27 13:01:16,163 - INFO - Initialization complete for project: BaldEagle
2024-09-27 13:01:16,163 - INFO - Plan entries: 2, Flow entries: 2, Unsteady entries: 1, Geometry entries: 1, Boundary conditions: 3
2024-09

Available plans in Bald Eagle Creek project:
  plan_number                                          full_path  \
0          01  c:\SCRATCH\ras-commander\examples\example_proj...   
1          03  c:\SCRATCH\ras-commander\examples\example_proj...   
2          04  c:\SCRATCH\ras-commander\examples\example_proj...   

  computation_interval dss_file flow_file friction_slope_method geom_file  \
0                15SEC      dss       u01                     1       g01   
1                10SEC      dss       u01                     1       g02   
2                10SEC      dss       u01                     1       g04   

  mapping_interval                                plan_title program_version  \
0             5MIN                  Unsteady Multi  9-SA run            5.00   
1             5MIN            Unsteady Run with 2D 50ft Grid            5.10   
2             5MIN  Unsteady Run with 2D 50ft User n Value R            5.10   

   ... run_unet run_wqnet     short_identifier      

2024-09-27 13:01:21,520 - INFO - HEC-RAS execution completed for plan: 05
2024-09-27 13:01:21,521 - INFO - Total run time for plan 05: 5.21 seconds
2024-09-27 13:01:21,612 - INFO - HEC-RAS execution completed for plan: 02
2024-09-27 13:01:21,612 - INFO - Total run time for plan 02: 5.30 seconds
2024-09-27 13:01:21,616 - WARNING - Results file for Plan number 02 does not exist.
2024-09-27 13:01:21,618 - WARNING - Results file for Plan number 05 does not exist.


Execution results:
Plan 02 execution: Successful
Plan 05 execution: Successful

Example 5: Getting results paths
No results found for Bald Eagle Creek plan 02
No results found for Muncie plan 05

Note: The original project folders can now be edited while the compute operations are running in separate folders.




In [25]:
# wait 10 seconds
time.sleep(10)

In [26]:
# Example 14 is long, and is a notebook.

In [27]:
print("Executing 15_plan_key_operations.py:")
%run 15_plan_key_operations.py
print("\n")

2024-09-27 13:01:31,650 - INFO - Example projects folder: c:\SCRATCH\ras-commander\examples\example_projects
2024-09-27 13:01:31,651 - INFO - Found zip file: c:\SCRATCH\ras-commander\examples\Example_Projects_6_5.zip
2024-09-27 13:01:31,651 - INFO - Loading project data from CSV...
2024-09-27 13:01:31,653 - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-09-27 13:01:31,653 - INFO - ----- RasExamples Extracting Project -----
2024-09-27 13:01:31,653 - INFO - Extracting project 'Balde Eagle Creek'
2024-09-27 13:01:31,653 - INFO - Project 'Balde Eagle Creek' already exists. Deleting existing folder...
2024-09-27 13:01:31,655 - INFO - Existing folder for project 'Balde Eagle Creek' has been deleted.
2024-09-27 13:01:31,681 - INFO - Successfully extracted project 'Balde Eagle Creek' to c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 13:01:31,682 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 

Executing 15_plan_key_operations.py:
Example 15: Getting and Setting Plan Keys
------------------------------------------
Working with Plan: 01

1. Current Plan Values:
  computation_interval: 2MIN
  simulation_date: 18FEB1999,0000,24FEB1999,0500
  short_identifier: UnsteadyFlow
  unet_d1_cores: None

2. Updating Plan Values:
  Updated computation_interval to: 30SEC
  Updated short_identifier to: Updated_Plan
  Updated unet_d1_cores to: 4

3. Verifying Updates:
  computation_interval: 30SEC
  short_identifier: Updated_Plan
  unet_d1_cores: None

4. Plan Description:
  Current description: None
  Updated description to: This is an updated plan description for Example 15.
  Verified updated description: This is an updated plan description for Example 15.

5. Handling Invalid Keys:

Example 15 completed.




In [28]:
# wait 10 seconds
time.sleep(10)

In [29]:
print("Executing 16_scanning_ras_project_info.py:")
%run 16_scanning_ras_project_info.py
print("\n")


2024-09-27 13:01:41,750 - INFO - Example projects folder: c:\SCRATCH\ras-commander\examples\example_projects
2024-09-27 13:01:41,751 - INFO - Found zip file: c:\SCRATCH\ras-commander\examples\Example_Projects_6_5.zip
2024-09-27 13:01:41,751 - INFO - Loading project data from CSV...
2024-09-27 13:01:41,752 - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-09-27 13:01:41,753 - INFO - Projects in category '1D Unsteady Flow Hydraulics': Balde Eagle Creek, Bridge Hydraulics, ContractionExpansionMinorLosses, Culvert Hydraulics, Culverts with Flap Gates, Dam Breaching, Elevation Controled Gates, Inline Structure with Gated Spillways, Internal Stage and Flow Boundary Condition, JunctionHydraulics, Lateral Strcuture with Gates, Lateral Structure connected to a River Reach, Lateral Structure Overflow Weir, Lateral Structure with Culverts, Lateral Structure with Culverts and Gates, Levee Breaching, Mixed Flow Regime, Multiple Reaches with Hydrau

Executing 16_scanning_ras_project_info.py:


2024-09-27 13:01:41,969 - INFO - Successfully extracted project 'Dam Breaching' to c:\SCRATCH\ras-commander\examples\example_projects\Dam Breaching
2024-09-27 13:01:41,970 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 13:01:41,970 - INFO - ----- RasExamples Extracting Project -----
2024-09-27 13:01:41,970 - INFO - Extracting project 'Elevation Controled Gates'
2024-09-27 13:01:41,980 - INFO - Successfully extracted project 'Elevation Controled Gates' to c:\SCRATCH\ras-commander\examples\example_projects\Elevation Controled Gates
2024-09-27 13:01:41,981 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 13:01:41,981 - INFO - ----- RasExamples Extracting Project -----
2024-09-27 13:01:41,981 - INFO - Extracting project 'Inline Structure with Gated Spillways'
2024-09-27 13:01:42,015 - INFO - Successfully extracted project 'Inline Structure with Gated Spillways' to c:\SCRATCH\ras-commander\examples\example_projects\Inline Structure with Gated Spillways
2024-0

Summary for category '1D Unsteady Flow Hydraulics' has been written to: c:\SCRATCH\ras-commander\examples\ras_example_categories\ras-commander 1D Unsteady Flow Hydraulics summary.txt


2024-09-27 13:01:44,021 - INFO - Successfully extracted project 'BaldEagleCrkMulti2D' to c:\SCRATCH\ras-commander\examples\example_projects\BaldEagleCrkMulti2D
2024-09-27 13:01:44,022 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 13:01:44,022 - INFO - ----- RasExamples Extracting Project -----
2024-09-27 13:01:44,022 - INFO - Extracting project 'Muncie'
2024-09-27 13:01:44,023 - INFO - Project 'Muncie' already exists. Deleting existing folder...
2024-09-27 13:01:44,031 - INFO - Existing folder for project 'Muncie' has been deleted.
2024-09-27 13:01:44,231 - INFO - Successfully extracted project 'Muncie' to c:\SCRATCH\ras-commander\examples\example_projects\Muncie
2024-09-27 13:01:44,232 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 13:01:44,234 - INFO - Initializing project in folder: c:\SCRATCH\ras-commander\examples\ras_example_categories\2D Unsteady Flow Hydraulics\BaldEagleCrkMulti2D
2024-09-27 13:01:44,234 - INFO - Using ras_instance with id: 19

Summary for category '2D Unsteady Flow Hydraulics' has been written to: c:\SCRATCH\ras-commander\examples\ras_example_categories\ras-commander 2D Unsteady Flow Hydraulics summary.txt
All category summaries have been generated.
Cleaned up original extracted example projects.




In [30]:
# wait 10 seconds
time.sleep(10)

In [31]:
print("Executing xx_edge_cases.py:")
%run xx_edge_cases.py
print("\n")

2024-09-27 13:01:54,361 - INFO - Example projects folder: c:\SCRATCH\ras-commander\examples\example_projects
2024-09-27 13:01:54,362 - INFO - Found zip file: c:\SCRATCH\ras-commander\examples\Example_Projects_6_5.zip
2024-09-27 13:01:54,363 - INFO - Loading project data from CSV...
2024-09-27 13:01:54,365 - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-09-27 13:01:54,365 - INFO - ----- RasExamples Extracting Project -----
2024-09-27 13:01:54,365 - INFO - Extracting project 'Balde Eagle Creek'
2024-09-27 13:01:54,391 - INFO - Successfully extracted project 'Balde Eagle Creek' to c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 13:01:54,391 - INFO - ----- RasExamples Extraction Complete -----
2024-09-27 13:01:54,392 - INFO - Initializing project in folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-09-27 13:01:54,392 - INFO - Using ras_instance with id: 140723289516752
2024

Executing xx_edge_cases.py:
Available plans:
  plan_number                                          full_path  \
0          01  c:\SCRATCH\ras-commander\examples\example_proj...   
1          02  c:\SCRATCH\ras-commander\examples\example_proj...   

  computation_interval dss_file flow_file friction_slope_method geom_file  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  mapping_interval                     plan_title program_version run_htab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  run_post_process run_sediment run_unet run_wqnet short_identifier  \
0                1            0        1         0     UnsteadyFlow   
1                1          NaN        1       NaN        SteadyRun   

                   simulation_date unet_d1_cores unet_use_existing_ib

2024-09-27 13:02:53,533 - INFO - HEC-RAS execution completed for plan: 01
2024-09-27 13:02:53,533 - INFO - Total run time for plan 01: 59.11 seconds
2024-09-27 13:02:53,536 - INFO - Successfully computed plan 01
2024-09-27 13:02:53,536 - INFO - Total run time for plan 01: 59.11 seconds
2024-09-27 13:02:53,536 - INFO - All selected plans have been executed.
2024-09-27 13:02:53,536 - INFO - compute_test_mode completed.
2024-09-27 13:02:53,550 - INFO - Copied project folder to destination: c:\SCRATCH\ras-commander\examples\example_projects\parallel_single_plan_result
2024-09-27 13:02:53,551 - INFO - Filtered plans to execute: ['01']
2024-09-27 13:02:53,552 - INFO - Adjusted max_workers to 1 based on the number of plans: 1
2024-09-27 13:02:53,566 - INFO - Created worker folder: c:\SCRATCH\ras-commander\examples\example_projects\parallel_single_plan_result [Worker 1]
2024-09-27 13:02:53,566 - INFO - Initializing project in folder: c:\SCRATCH\ras-commander\examples\example_projects\parallel_

Execution of plan 01 completed using compute_test_mode

Example 2: Executing a single plan using compute_parallel


2024-09-27 13:03:52,321 - INFO - HEC-RAS execution completed for plan: 01
2024-09-27 13:03:52,321 - INFO - Total run time for plan 01: 58.74 seconds
2024-09-27 13:03:52,323 - INFO - Plan 01 executed in worker 1
2024-09-27 13:03:52,324 - INFO - Final destination for computed results: c:\SCRATCH\ras-commander\examples\example_projects\parallel_single_plan_result
2024-09-27 13:03:52,325 - DEBUG - Moved c:\SCRATCH\ras-commander\examples\example_projects\parallel_single_plan_result [Worker 1]\BaldEagle.b01 to c:\SCRATCH\ras-commander\examples\example_projects\parallel_single_plan_result
2024-09-27 13:03:52,326 - DEBUG - Moved c:\SCRATCH\ras-commander\examples\example_projects\parallel_single_plan_result [Worker 1]\BaldEagle.bco01 to c:\SCRATCH\ras-commander\examples\example_projects\parallel_single_plan_result
2024-09-27 13:03:52,326 - DEBUG - Removed existing file at c:\SCRATCH\ras-commander\examples\example_projects\parallel_single_plan_result\BaldEagle.c01
2024-09-27 13:03:52,327 - DEBUG

Parallel execution of single plan results:
Plan 01: Successful

Example 3: Executing a single plan using compute_test_mode with a string input


2024-09-27 13:03:56,991 - INFO - HEC-RAS execution completed for plan: 02
2024-09-27 13:03:56,991 - INFO - Total run time for plan 02: 4.59 seconds
2024-09-27 13:03:56,993 - INFO - Successfully computed plan 02
2024-09-27 13:03:56,993 - INFO - Total run time for plan 02: 4.60 seconds
2024-09-27 13:03:56,993 - INFO - All selected plans have been executed.
2024-09-27 13:03:56,993 - INFO - compute_test_mode completed.
2024-09-27 13:03:57,007 - INFO - Copied project folder to destination: c:\SCRATCH\ras-commander\examples\example_projects\parallel_single_plan_string_result
2024-09-27 13:03:57,008 - INFO - Filtered plans to execute: ['01']
2024-09-27 13:03:57,009 - INFO - Adjusted max_workers to 1 based on the number of plans: 1
2024-09-27 13:03:57,022 - INFO - Created worker folder: c:\SCRATCH\ras-commander\examples\example_projects\parallel_single_plan_string_result [Worker 1]
2024-09-27 13:03:57,023 - INFO - Initializing project in folder: c:\SCRATCH\ras-commander\examples\example_projec

Execution of plan 02 completed using compute_test_mode with string input

Example 4: Executing a single plan using compute_parallel with a string input


2024-09-27 13:04:56,133 - INFO - HEC-RAS execution completed for plan: 01
2024-09-27 13:04:56,134 - INFO - Total run time for plan 01: 59.10 seconds
2024-09-27 13:04:56,136 - INFO - Plan 01 executed in worker 1
2024-09-27 13:04:56,136 - INFO - Final destination for computed results: c:\SCRATCH\ras-commander\examples\example_projects\parallel_single_plan_string_result
2024-09-27 13:04:56,137 - DEBUG - Moved c:\SCRATCH\ras-commander\examples\example_projects\parallel_single_plan_string_result [Worker 1]\BaldEagle.b01 to c:\SCRATCH\ras-commander\examples\example_projects\parallel_single_plan_string_result
2024-09-27 13:04:56,138 - DEBUG - Moved c:\SCRATCH\ras-commander\examples\example_projects\parallel_single_plan_string_result [Worker 1]\BaldEagle.bco01 to c:\SCRATCH\ras-commander\examples\example_projects\parallel_single_plan_string_result
2024-09-27 13:04:56,139 - DEBUG - Removed existing file at c:\SCRATCH\ras-commander\examples\example_projects\parallel_single_plan_string_result\Bal

Parallel execution of single plan (string input) results:
Plan 01: Successful

Example 5: Attempting to execute with an empty plan list


2024-09-27 13:05:54,885 - INFO - HEC-RAS execution completed for plan: 01
2024-09-27 13:05:54,885 - INFO - Total run time for plan 01: 58.68 seconds
2024-09-27 13:05:54,887 - INFO - Successfully computed plan 01
2024-09-27 13:05:54,888 - INFO - Total run time for plan 01: 58.69 seconds
2024-09-27 13:05:54,888 - INFO - Using ras_object with project folder: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [EmptyPlanList]
2024-09-27 13:05:54,890 - INFO - Plan file for Plan number 02 found at: c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [EmptyPlanList]\BaldEagle.p02
2024-09-27 13:05:54,890 - INFO - Running HEC-RAS from the Command Line:
2024-09-27 13:05:54,891 - INFO - Running command: "C:\Program Files (x86)\HEC\HEC-RAS\6.5\Ras.exe" -c "C:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [EmptyPlanList]\BaldEagle.prj" "c:\SCRATCH\ras-commander\examples\example_projects\Balde Eagle Creek [EmptyPlanList]\BaldEagle.p02"
2024-09-27


Example 6: Attempting to execute with a non-existent plan number





# Guide for Locating Relevant Examples:

## Category: 1D Unsteady Flow Hydraulics

### Flow Hydrograph

| Example | Plan File |
|---------|-----------|
| Balde Eagle Creek | BaldEagle.p01, BaldEagle.p02 |
| ContractionExpansionMinorLosses | ContExp.p01, ContExp.p02, ContExp.p03, ContExp.p04 |
| Culvert Hydraulics | Beav_Culvert.p01 |
| Culverts with Flap Gates | CulvertFlap.p01 |
| Dam Breaching | BaldEagleDamBrk.p07, BaldEagleDamBrk.p02, BaldEagleDamBrk.p10, BaldEagleDamBrk.p06, BaldEagleDamBrk.p08, BaldEagleDamBrk.p04, BaldEagleDamBrk.p03, BaldEagleDamBrk.p05, BaldEagleDamBrk.p09, BaldEagleDamBrk.p11 |
| Elevation Controled Gates | ElevControledGates.p01 |
| Inline Structure with Gated Spillways | Inline_3Gates.p01 |
| Internal Stage and Flow Boundary Condition | IBStageFlowTest.p01 |
| JunctionHydraulics | JunctionHydraulics.p01, JunctionHydraulics.p02, JunctionHydraulics.p04, JunctionHydraulics.p03 |
| Lateral Strcuture with Gates | TwoGateGrps.p01 |
| Lateral Structure connected to a River Reach | 3Reach_with_Lat.p01 |
| Lateral Structure Overflow Weir | LatMultXS.p01 |
| Lateral Structure with Culverts | LatWeirCulverts.p01 |
| Lateral Structure with Culverts and Gates | Beav_Lateral.p01 |
| Levee Breaching | LeveeBreach.p01 |
| Multiple Reaches with Hydraulic Structures | 3ReachUnsteady.p01, 3ReachUnsteady.p02 |
| Pumping Station | Pumps.p01 |
| Pumping Station with Rules | PumpRule.p03 |
| Rule Operations | Canal.p01 |
| Simplified Physical Breaching | BaldEDmbrk.p01 |
| UngagedAreaInflows | Ungaged_area.p03, Ungaged_area.p02 |
| Unsteady Flow Encroachment Analysis | UnsteadyEncroachment.p03, UnsteadyEncroachment.p02 |

### Normal Depth

| Example | Plan File |
|---------|-----------|
| Balde Eagle Creek | BaldEagle.p01, BaldEagle.p02 |
| ContractionExpansionMinorLosses | ContExp.p01, ContExp.p02, ContExp.p03, ContExp.p04 |
| Culvert Hydraulics | Beav_Culvert.p01 |
| Dam Breaching | BaldEagleDamBrk.p07, BaldEagleDamBrk.p02, BaldEagleDamBrk.p10, BaldEagleDamBrk.p06, BaldEagleDamBrk.p08, BaldEagleDamBrk.p04, BaldEagleDamBrk.p03, BaldEagleDamBrk.p05, BaldEagleDamBrk.p09, BaldEagleDamBrk.p11 |
| JunctionHydraulics | JunctionHydraulics.p01, JunctionHydraulics.p02, JunctionHydraulics.p04, JunctionHydraulics.p03 |
| Lateral Structure connected to a River Reach | 3Reach_with_Lat.p01 |
| Multiple Reaches with Hydraulic Structures | 3ReachUnsteady.p01, 3ReachUnsteady.p02 |
| Rule Operations | Canal.p01 |
| Simplified Physical Breaching | BaldEDmbrk.p01 |
| UngagedAreaInflows | Ungaged_area.p03, Ungaged_area.p02 |

### Gate Opening

| Example | Plan File |
|---------|-----------|
| Balde Eagle Creek | BaldEagle.p01, BaldEagle.p02 |
| ContractionExpansionMinorLosses | ContExp.p01, ContExp.p02, ContExp.p03, ContExp.p04 |
| Dam Breaching | BaldEagleDamBrk.p07, BaldEagleDamBrk.p02, BaldEagleDamBrk.p10, BaldEagleDamBrk.p06, BaldEagleDamBrk.p08, BaldEagleDamBrk.p04, BaldEagleDamBrk.p03, BaldEagleDamBrk.p05, BaldEagleDamBrk.p09, BaldEagleDamBrk.p11 |
| Elevation Controled Gates | ElevControledGates.p01 |
| Inline Structure with Gated Spillways | Inline_3Gates.p01 |
| Lateral Strcuture with Gates | TwoGateGrps.p01 |
| Lateral Structure with Culverts and Gates | Beav_Lateral.p01 |
| Simplified Physical Breaching | BaldEDmbrk.p01 |

### Stage Hydrograph

| Example | Plan File |
|---------|-----------|
| Internal Stage and Flow Boundary Condition | IBStageFlowTest.p01 |
| Pumping Station with Rules | PumpRule.p03 |
| Rule Operations | Canal.p01 |

### Lateral Inflow Hydrograph

| Example | Plan File |
|---------|-----------|
| Dam Breaching | BaldEagleDamBrk.p07, BaldEagleDamBrk.p02, BaldEagleDamBrk.p10, BaldEagleDamBrk.p06, BaldEagleDamBrk.p08, BaldEagleDamBrk.p04, BaldEagleDamBrk.p03, BaldEagleDamBrk.p05, BaldEagleDamBrk.p09, BaldEagleDamBrk.p11 |
| Multiple Reaches with Hydraulic Structures | 3ReachUnsteady.p01, 3ReachUnsteady.p02 |
| Pumping Station | Pumps.p01 |
| Rule Operations | Canal.p01 |
| UngagedAreaInflows | Ungaged_area.p03, Ungaged_area.p02 |

### Uniform Lateral Inflow Hydrograph

| Example | Plan File |
|---------|-----------|
| Dam Breaching | BaldEagleDamBrk.p07, BaldEagleDamBrk.p02, BaldEagleDamBrk.p10, BaldEagleDamBrk.p06, BaldEagleDamBrk.p08, BaldEagleDamBrk.p04, BaldEagleDamBrk.p03, BaldEagleDamBrk.p05, BaldEagleDamBrk.p09, BaldEagleDamBrk.p11 |
| UngagedAreaInflows | Ungaged_area.p03, Ungaged_area.p02 |

### Unknown or Special Conditions

| Example | Plan File |
|---------|-----------|
| Culverts with Flap Gates | CulvertFlap.p01 |
| Elevation Controled Gates | ElevControledGates.p01 |
| Internal Stage and Flow Boundary Condition | IBStageFlowTest.p01 |
| Lateral Structure Overflow Weir | LatMultXS.p01 |
| Lateral Structure with Culverts | LatWeirCulverts.p01 |
| Levee Breaching | LeveeBreach.p01 |
| Pumping Station with Rules | PumpRule.p03 |
| UngagedAreaInflows | Ungaged_area.p03, Ungaged_area.p02 |
| Unsteady Flow Encroachment Analysis | UnsteadyEncroachment.p03, UnsteadyEncroachment.p02 |

## Category: 2D Unsteady Flow Hydraulics

# HEC-RAS Unsteady Flow Examples Analysis

## Version 6.30

### Example: Muncie

| Boundary Condition Types | Plan Files |
|--------------------------|------------|
| Flow Hydrograph          | Muncie.p01 (Unsteady Multi  9-SA run) |
| Normal Depth             | Muncie.p03 (Unsteady Run with 2D 50ft Grid) |
|                          | Muncie.p04 (Unsteady Run with 2D 50ft User n Value R) |

| Project | Version | Boundary Conditions | Plans |
|---------|---------|---------------------|-------|
| Muncie  | 6.30    | - Flow Hydrograph (White River, Muncie, Station 15696.24) | 1. Muncie.p01: Unsteady Multi 9-SA run |
|         |         | - Normal Depth (White River, Muncie, Station 237.6455)    | 2. Muncie.p03: Unsteady Run with 2D 50ft Grid |
|         |         |                                                           | 3. Muncie.p04: Unsteady Run with 2D 50ft User n Value R |

## Version 6.00

### Example: BaldEagleCrkMulti2D

| Boundary Condition Types | Plan Files |
|--------------------------|------------|
| Flow Hydrograph          | BaldEagleDamBrk.p06 (Gridded Precip - Infiltration) |
| Gate Opening             |  |
| Lateral Inflow Hydrograph |  |
| Uniform Lateral Inflow Hydrograph |  |
| Normal Depth             |  |

| Project | Version | Boundary Conditions | Plans |
|---------|---------|---------------------|-------|
| BaldEagleCrkMulti2D | 6.00 | - Flow Hydrograph | 1. BaldEagleDamBrk.p06: Gridded Precip - Infiltration |
|                     |      | - Gate Opening |  |
|                     |      | - Lateral Inflow Hydrograph |  |
|                     |      | - Uniform Lateral Inflow Hydrograph |  |
|                     |      | - Normal Depth |  |

## Version 5.10

### Example: BaldEagleCrkMulti2D

| Boundary Condition Types | Plan Files |
|--------------------------|------------|
| Flow Hydrograph          | BaldEagleDamBrk.p13 (PMF with Multi 2D Areas) |
| Gate Opening             | BaldEagleDamBrk.p15 (1d-2D Dambreak Refined Grid) |
| Lateral Inflow Hydrograph | BaldEagleDamBrk.p02 (SA to Detailed 2D Breach) |
| Uniform Lateral Inflow Hydrograph | BaldEagleDamBrk.p05 (Single 2D area with Bridges FEQ) |
| Normal Depth             | BaldEagleDamBrk.p01 (SA to Detailed 2D Breach FEQ) |

| Project | Version | Boundary Conditions | Plans |
|---------|---------|---------------------|-------|
| BaldEagleCrkMulti2D | 5.10 | - Flow Hydrograph | 1. BaldEagleDamBrk.p13: PMF with Multi 2D Areas |
|                     |      | - Gate Opening | 2. BaldEagleDamBrk.p15: 1d-2D Dambreak Refined Grid |
|                     |      | - Lateral Inflow Hydrograph | 3. BaldEagleDamBrk.p02: SA to Detailed 2D Breach |
|                     |      | - Uniform Lateral Inflow Hydrograph | 4. BaldEagleDamBrk.p05: Single 2D area with Bridges FEQ |
|                     |      | - Normal Depth | 5. BaldEagleDamBrk.p01: SA to Detailed 2D Breach FEQ |

## Version 5.00

### Example: BaldEagleCrkMulti2D

| Boundary Condition Types | Plan Files |
|--------------------------|------------|
| Flow Hydrograph          | BaldEagleDamBrk.p17 (2D to 1D No Dam) |
| Gate Opening             | BaldEagleDamBrk.p18 (2D to 2D Run) |
| Lateral Inflow Hydrograph | BaldEagleDamBrk.p19 (SA to 2D Dam Break Run) |
| Uniform Lateral Inflow Hydrograph | BaldEagleDamBrk.p03 (Single 2D Area - Internal Dam Structure) |
| Normal Depth             | BaldEagleDamBrk.p04 (SA to 2D Area Conn - 2D Levee Structure) |

| Project | Version | Boundary Conditions | Plans |
|---------|---------|---------------------|-------|
| BaldEagleCrkMulti2D | 5.00 | - Flow Hydrograph | 1. BaldEagleDamBrk.p17: 2D to 1D No Dam |
|                     |      | - Gate Opening | 2. BaldEagleDamBrk.p18: 2D to 2D Run |
|                     |      | - Lateral Inflow Hydrograph | 3. BaldEagleDamBrk.p19: SA to 2D Dam Break Run |
|                     |      | - Uniform Lateral Inflow Hydrograph | 4. BaldEagleDamBrk.p03: Single 2D Area - Internal Dam Structure |
|                     |      | - Normal Depth | 5. BaldEagleDamBrk.p04: SA to 2D Area Conn - 2D Levee Structure |

### Need the same tables for geometry key types

Also need to open and save all example files in 6.5 using HECRASController (making a function to do this)

This can be used to help find examples if the user wants to do something that the library is currently not capable of.  

Using the RasExamples class, the user can extract these files and inspect them to find specific examples.  